cognitive/src/test/scala/com/microsoft/azure/synapse/ml/cognitive/anomaly/MultivariateAnamolyDetectionSuite.scala

```diff
@@ -264,3 +270,67 @@ class SimpleFitMultivariateAnomalySuite extends EstimatorFuzzing[SimpleFitMultiv
 
   override def modelReader: MLReadable[_] = SimpleDetectMultivariateAnomaly
 }
+
+class DetectLastMultivariateAnomalySuite extends TransformerFuzzing[DetectLastMultivariateAnomaly]
+  with MADTestUtils {
+
+  val hc: Configuration = spark.sparkContext.hadoopConfiguration
+  hc.set("fs.azure", "org.apache.hadoop.fs.azure.NativeAzureFileSystem")
+  hc.set(s"fs.azure.account.keyprovider.$storageAccount.blob.core.windows.net",
+    "org.apache.hadoop.fs.azure.SimpleKeyProvider")
+  hc.set(s"fs.azure.account.key.$storageAccount.blob.core.windows.net", storageKey)
+
+  val sfma: SimpleFitMultivariateAnomaly = new SimpleFitMultivariateAnomaly()
+    .setSubscriptionKey(anomalyKey)
+    .setLocation(anomalyLocation)
+    .setOutputCol("result")
+    .setStartTime(startTime)
+    .setEndTime(endTime)
+    .setIntermediateSaveDir(intermediateSaveDir)
+    .setTimestampCol(timestampColumn)
+    .setInputCols(inputColumns)
+    .setSlidingWindow(50)
+
+  val model: SimpleDetectMultivariateAnomaly = sfma.fit(df)
+  val modelId: String = model.getModelId
+
+  MADUtils.CreatedModels += modelId
+
+  lazy val dlma: DetectLastMultivariateAnomaly = new DetectLastMultivariateAnomaly()
+    .setSubscriptionKey(anomalyKey)
+    .setLocation(anomalyLocation)
+    .setModelId(modelId)
+    .setInputVariablesCols(inputColumns)
+    .setOutputCol("result")
+    .setTimestampCol(timestampColumn)
+
+  test("Basic Usage") {
+    val result = dlma.setBatchSize(50)
+      .transform(df.limit(100))
+      .collect()
+    assert(result(0).get(6) == null)
+    assert(!result(50).getAs[Boolean]("isAnomaly"))
+    assert(result(68).getAs[Boolean]("isAnomaly"))
+  }
+
+  test("Error if batch size is smaller than sliding window") {
+    val result = dlma.setBatchSize(10).transform(df.limit(50))
+    result.show(50, truncate = false)
+    assert(result.collect().head.getAs[StringType](dlma.getErrorCol).toString.contains("NotEnoughData"))
+  }
```

Comment:
Can you provide more details on why this is a failure? Im a little confused about whats going on here

Reply:
Because the number of data points we use to infer should be larger than sliding window, say your sliding window is 30, then you could only use the first 30 data points to infer the 31st data point. In this test case, batchSize is 10, but the model sfma has sliding window 50 (.setSlidingWindow(50)), so it couldn't infer any datapoints using 10 data points.